packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -857,6 +857,15 @@ class _InkResponseState extends State<_InkResponseStateWidget>
       statesController.update(MaterialState.disabled, !enabled);
       if (!enabled) {
         statesController.update(MaterialState.pressed, false);
+        // Remove the existing hover highlight immediately when enabled is false.
+        // Do not rely on updateHighlight or InkHighlight.deactivate to not break
+        // the expected lifecycle which is updating _hovering when the mouse exit.
+        // Manually updating _hovering here or calling InkHighlight.deactivate
+        // will lead to onHover not being called or call when it is not allowed.
+        final InkHighlight? hoverHighLight = _highlights[_HighlightType.hover];
```

Comment:
```suggestion
        final InkHighlight? hoverHighlight = _highlights[_HighlightType.hover];
```
Since highlight is just one word.